src/tasks/mutemusic.py

```diff
@@ -0,0 +1,243 @@
+import os, time, pandas
+
+from psychopy import prefs
+prefs.hardware['audioLib'] = ['sounddevice']
+from psychopy import visual, sound, event, core, logging
+
+from .task_base import Task
+from ..shared import config, utils
+from ..shared.eyetracking import fixation_dot
+
+#task : 1 run = 1 playlist = around 10 audio tracks
+#repeat for n songs in subXX_runXX.csv :
+#   step 1: DEFAULT_INSTRUCTION
+#   step 2: display audio
+#   step 3: Auditory imagery assessment
+#   step 4: Familiarity assessment
+
+#Global Variables if multiples tasks
+INSTRUCTION_DURATION = 3
+DEFAULT_INSTRUCTION = """Listen to the following tracks"""
+AUDITORY_IMAGERY_ASSESSMENT = ("During the silences, did you imagine the missing part of the music clip you’ve heard ?", ['No', '', 'Partially', '', 'Yes'])
+
+class Playlist(Task):
+#Derived from SoundTaskBase (Narratives task)
+    def __init__(self, tsv_path, initial_wait=2, question_duration = 5, **kwargs):
+        super().__init__(**kwargs)
+
+        if not os.path.exists(tsv_path):
+            raise ValueError("File %s does not exists" % tsv_path)   
+        else :
+            self.tsv_path = tsv_path
+            file = open(tsv_path, "r")
+            self.playlist = pandas.read_table(file, sep='\t')
+            file.close()
+
+        self.initial_wait = initial_wait
+        self.question_duration = question_duration
+        self.instruction = DEFAULT_INSTRUCTION
+    
+    def _instructions(self, exp_win, ctl_win):
+        print(self.instruction)
+        screen_text = visual.TextStim(
+            exp_win,
+            text=self.instruction,
+            alignText="center",
+            color="white",
+            units='height',
+            height=0.05
+        )
+        screen_text.draw(exp_win)
+        if ctl_win:
+            screen_text.draw(ctl_win)
+        yield True
+        time.sleep(INSTRUCTION_DURATION)
+        yield True
+
+    def _setup(self, exp_win):
+        super()._setup(exp_win)
+        self.fixation = fixation_dot(exp_win)
+
+    def _handle_controller_presses(self, ISI):
+        ISI.start(0.01)
+        self._new_key_pressed = event.getKeys('lra')
+        ISI.complete()
+
+    def _questionnaire(self, exp_win, ctl_win, question, answers):
+        event.getKeys('lra') # flush keys
+        self.ISI = core.StaticPeriod(win=exp_win)
```

Comment:
Any reason for the use of that StaticPeriod.
It is giving me warnings that timing wasn't respected.
If I remove it (here and in the controller press func), it works the same.

Reply:
all ISI related lines in  def _handle_controller_presses and def_questionnaire (64, 66, 70) are here because of environment compatibility issues on my side, but for you / mri you can definitely remove it.